app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUiRenderer.kt

```diff
@@ -11,18 +11,17 @@ class CustomDrugEntryUiRenderer(
     if (!model.isCustomDrugEntrySheetInfoLoaded) {
       ui.showProgressBar()
       ui.hideCustomDrugEntryUi()
-      return
-    }
-
-    ui.hideProgressBar()
-    ui.showCustomDrugEntryUi()
+    } else {
+      ui.hideProgressBar()
+      ui.showCustomDrugEntryUi()
 
-    initialSetup(model.openAs)
+      initialSetup(model.openAs)
 
-    if (model.drugFrequencyToFrequencyChoiceItemMap != null)
-      setSheetTitle(model.drugName, model.dosage, model.drugFrequencyToFrequencyChoiceItemMap[model.frequency]!!.label)
+      if (model.drugFrequencyToFrequencyChoiceItemMap != null)
+        setSheetTitle(model.drugName, model.dosage, model.drugFrequencyToFrequencyChoiceItemMap[model.frequency]!!.label)
```

Comment:
Just an FYI, we tend to avoid using nested conditionals and extract them to new a function.